functions-eventbrite.php

```diff
@@ -1,121 +1,220 @@
 <?php
+
 /**
  * Eventbrite API
  *
  */
 
-function tna_ebapi_url( $organiser, $category, $token ) {
-
-	$url = 'https://www.eventbriteapi.com/v3/events/search/?sort_by=date&organizer.id='.$organiser.$category.'&token='.$token.'&expand=ticket_classes';
+class Simple_Eventbrite_List {
 
-	return $url;
-}
+	/**
+	 * @param $organiser
+	 * @param $category
+	 * @param $token
+	 *
+	 * @return string
+	 */
+	public function url( $organiser, $category, $token ) {
 
-function tna_ebapi_get_eventbrite_json( $url ) {
+		$url = 'https://www.eventbriteapi.com/v3/events/search/?sort_by=date&organizer.id=' . $organiser . $category . '&token=' . $token . '&expand=ticket_classes';
 
-	if ( !class_exists('WP_Http') ) {
-		include_once( ABSPATH . WPINC . '/class-http.php');
+		return $url;
 	}
 
-	$request = new WP_Http;
-	$result = $request->request( $url );
+	/**
+	 * @param $result
+	 *
+	 * @return bool
+	 */
+	public function check_result( $result ) {
+
```

Comment:
**Warning: Undefined functions.**
When dealing with Wordpress functions and custom functions it's a good habit to wrap the whole statements with if (function_exists(''')) around WP functions. This way you tell the IDE that you are in control and you know what you are doing as well as PHP unit won't complain when trying to test particular things from within your custom function. If that makes sense? If not please let me know.

e.g. functions like (is_wp_error, wp_remote_retrieve_response_code)